### PR TITLE
Redmine #9747: Fixes malformed JSON

### PR DESCRIPTION
**Pull request by s-fiebig**
Fixes malformed JSON by escaping backslash in case of a domain prefix.

- [x] Redmine Issue: https://redmine.pfsense.org/issues/9747
- [x] Ready for review

### CODE REVIEW

**Comment by s-fiebig** [comment]
This would be definitely the better solution. Will have a look at that, as soon as I get the time.

**Comment by jim-p** [comment]
Looks like you need to initialize `$data` before its first use or it returns an error and does not render the widget properly.

**Comment by s-fiebig** [comment]
It's a warning and it did actually render the widget at my pfsense, but of course I will change it.

File: src/usr/local/www/widgets/widgets/ipsec.widget.php [hunk 1]
```diff
@@ -125,58 +125,48 @@
...
-			$jsondata .= "<td>" . $ipsec['remote-subnet'] . "<br />(" . htmlspecialchars($ipsec['dest']) . ")</td>";
-			$jsondata .= "<td>" . htmlspecialchars($ipsec['descr']) . "</td>";
-
+			$data->tunnel .= "<tr>";
+			$data->tunnel .= "<td>" . htmlspecialchars($ipsec['src']) . "</td>";
+			$data->tunnel .= "<td>" . $ipsec['remote-subnet'] . "<br />(" . htmlspecialchars($ipsec['dest']) . ")</td>";
+			$data->tunnel .= "<td>" . htmlspecialchars($ipsec['descr']) . "</td>";
 			if ($ipsec['status'] == "true") {
-				$jsondata .= '<td><i class=\"fa fa-arrow-up text-success\"></i></td>';
+				$data->tunnel .= '<td><i class=\"fa fa-arrow-up text-success\"></i></td>';
```
Comment (jim-p) [inline src/usr/local/www/widgets/widgets/ipsec.widget.php]: These double quotes no longer need escaped, they are getting escaped again which breaks the status icon.

File: src/usr/local/www/widgets/widgets/ipsec.widget.php [hunk 2]
```diff
@@ -125,58 +125,48 @@
...
 			} else {
-				$jsondata .= '<td><i class=\"fa fa-arrow-down text-danger\"></i></td>';
+				$data->tunnel .= '<td><i class=\"fa fa-arrow-down text-danger\"></i></td>';
```
Comment (jim-p) [inline src/usr/local/www/widgets/widgets/ipsec.widget.php]: These double quotes no longer need escaped, they are getting escaped again which breaks the status icon.

**Comment by s-fiebig** [comment]
Thanks for pointing this out! This was totally under my radar, because my current setup has no tunnels.

**Pull request merged**